$OpenBSD$

 - Allow overide of includes and libs.
 - Fix DESTDIR clash
 - Honour CFLAGS

--- Makefile.orig	Fri Feb 24 08:17:29 2012
+++ Makefile	Mon Jul  2 16:32:01 2012
@@ -2,20 +2,20 @@
 RANLIB= ranlib
 
 OCAML_LIBDIR:= $(shell ocamlc -where)
-GMP_INCLUDES= -I/opt/gmp-4.1.2/include -I/users/absint2/local/include -I$(HOME)/packages/gmp/include
+GMP_INCLUDES?= -I/opt/gmp-4.1.2/include -I/users/absint2/local/include -I$(HOME)/packages/gmp/include
 
-GMP_LIBDIR=/opt/gmp-4.1.2/lib
-DESTDIR= $(OCAML_LIBDIR)/gmp
+GMP_LIBDIR?=/opt/gmp-4.1.2/lib
+INSTDIR= ${DESTDIR}/$(OCAML_LIBDIR)/gmp
 
 #RLIBFLAGS= -cclib "-Wl,-rpath $(GMP_LIBDIR)" # Linux, FreeBSD
 #RLIBFLAGS= -cclib "-Wl,-R $(GMP_LIBDIR)" # Solaris
 # RLIBFLAGS= # MacOS X
 
 LIBFLAGS= -cclib -L. -cclib -L$(GMP_LIBDIR) $(RLIBFLAGS) \
-	-cclib -lmpfr -cclib -lgmp -cclib -L$(DESTDIR)
+	-cclib -lmpfr -cclib -lgmp -cclib -L${OCAML_LIBDIR}
 
 #CC= icc
-CFLAGS_MISC= -Wall -Wno-unused -Werror -g -O3
+CFLAGS_MISC?= -Wall -Wno-unused -Werror -g -O3
 #CFLAGS_MISC=
 CFLAGS_INCLUDE= -I $(OCAML_LIBDIR) $(GMP_INCLUDES)
 CFLAGS= $(CFLAGS_MISC) $(CFLAGS_INCLUDE)
@@ -36,8 +36,8 @@ TESTS= test_suite test_suite.opt
 all:	$(LIBS) tests
 
 install: all
-	-mkdir $(DESTDIR)
-	cp $(LIBS) gmp.mli $(DESTDIR)
+	-mkdir -p $(INSTDIR)
+	cp $(LIBS) gmp.mli $(INSTDIR)
 
 tests:	$(LIBS) $(TESTS)
 	./test_suite
